CampusDelSaber/hermes
# Feature/ id 176/ Routes algorithm

Logic of the algorithm for the calculation of the route, without the influence of traffic or incidents was implemented.

- [X]  Apply Dijkstra algorithm
- [X] Represent a node as a street corner
- [X] The Edge will be the direction of a street to other corner, it can be unidirectional or bidirectional
- [X] Be able to calculate the distances from two coordinates
- [X] The algorithm should return what was decided. https://tree.taiga.io/project/marcel-morales-hermes/wiki/input-output-navigation-algorithm
- [X] Graph classes creation

**Steps to test**
- `Run DijkstraAlgorithmRealTest`
- Look at the console
- Search for the log that contains the GeoJSON as string
- 
![image](https://github.com/CampusDelSaber/hermes/assets/135616751/c68bef40-8d27-41f8-9292-c5ecb313a7b5)
- Copy all the json and select the GeoJSON string of Route A (the shortest route)
- 
![image](https://github.com/CampusDelSaber/hermes/assets/135616751/f90ea969-d8a1-457a-811a-c016b9fa9e04)
- Paste it in http://geojson.io/#map=17.2/-17.394824/-66.155188
- 
![image](https://github.com/CampusDelSaber/hermes/assets/135616751/583b601f-a3b2-4b13-82e4-08d07763619b)

- Do same steps with Route B and check that the source and destination point are the same, check the street directions and check that Route A distance is shorter (There is a "distance" key in the GeoJson in km. to check it).
- Route B:
- 
![image](https://github.com/CampusDelSaber/hermes/assets/135616751/9fbc6ad7-7372-4436-afca-0bd4154d86e6)
